src/pymor/vectorarrays/interface.py

```diff
@@ -577,16 +577,23 @@ def check_ind_unique(self, ind):
     def len_ind(self, ind):
         """Return the number of given indices."""
         l = len(self)
-        return (len(range(*ind.indices(l))) if type(ind) is slice else
-                1 if not hasattr(ind, '__len__') else
-                len(ind))
+        if type(ind) is slice:
+            return len(range(*ind.indices(l)))
+        try:
+            return len(ind)
+        except TypeError:
+            return 1
```

Comment:
Why these changes?

Reply:
More pythonic and better readable than hastattr.

Reply:
I think that's arguable, but ok.